hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -0,0 +1,422 @@
+                instance -> (EventJournalReader<EventJournalMapEvent>) instance.getMap(mapName),
+                predicate, projection, newestSequence);
+    }
+
+    public static <T> ProcessorMetaSupplier streamCache(String cacheName,
+                                                        DistributedPredicate<EventJournalCacheEvent> predicate,
+                                                        DistributedFunction<EventJournalCacheEvent, T> projection,
+                                                        boolean newestSequence) {
+        return new LocalClusterMetaSupplier<>(
+                instance -> (EventJournalReader<EventJournalCacheEvent>) instance.getCacheManager().getCache(cacheName),
+                predicate, projection, newestSequence);
+    }
+
+    public static <T> ProcessorMetaSupplier streamCache(String cacheName,
+                                                        ClientConfig clientConfig,
+                                                        DistributedPredicate<EventJournalCacheEvent> predicate,
+                                                        DistributedFunction<EventJournalCacheEvent, T> projection,
+                                                        boolean newestSequence) {
+        return new RemoteClusterMetaSupplier<>(clientConfig,
+                instance -> (EventJournalReader<EventJournalCacheEvent>) instance.getCacheManager().getCache(cacheName),
+                predicate, projection, newestSequence);
+    }
+
+    @Override
+    protected void init(@Nonnull Context context) throws Exception {
+        super.init(context);
+        jobFuture = context.jobFuture();
+    }
+
+    private void subscribe(List<Integer> partitions) {
+        List<ICompletableFuture<EventJournalInitialSubscriberState>> futures = new ArrayList<>();
+        try {
+            for (Integer partitionId : partitions) {
+                futures.add(eventJournalReader.subscribeToEventJournal(partitionId));
+            }
+            for (int i = 0; i < futures.size(); i++) {
+                ICompletableFuture<EventJournalInitialSubscriberState> future = futures.get(i);
+                offsetMap.put(partitions.get(i), getSequence(future.get()));
+            }
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
+        }
+    }
+
+    private long getSequence(EventJournalInitialSubscriberState state) {
+        if (newestSequence) {
+            return state.getNewestSequence() == -1 ? 0 : state.getNewestSequence();
+        }
+        return state.getOldestSequence();
+    }
+
+    @Override
+    public boolean complete() {
+        long idleCount = 0;
+        while (!jobFuture.isDone()) {
+            boolean fetched = poll(this::emit);
+            if (fetched) {
+                idleCount = 0;
+            } else {
+                IDLER.idle(++idleCount);
+            }
+        }
+        return true;
+    }
+
+    private boolean poll(Consumer<T> consumer) {
```

Comment:
what's the use of `Consumer` here? you can just call `emit` directly.. as there is ever only one consumer